src/api/api_context.cpp

```diff
@@ -81,14 +82,12 @@ namespace api {
     void context::flush_objects() {
 #ifndef SINGLE_THREAD
         if (!m_concurrent_dec_ref)
-            return;        
-        {
-            lock_guard lock(m_mux);
-            if (m_asts_to_flush.empty() && m_objects_to_flush.empty())
-                return;
-            m_asts_to_flush2.swap(m_asts_to_flush);
-            m_objects_to_flush2.swap(m_objects_to_flush);
-        }
+            return;
+        lock_guard lock(m_mux);
```

Comment:
global lock guard allows contention during dec_ref

Reply:
Yes, I much prefer that to random segfaults though!